Develop aws sdk2 update

telus-terchu opened the pull request:
## Motivation

Why am I doing this

## Modification

What did I change

## PR Checklist

- [x] been self-reviewed.
- [ ] Added javadocs for most classes and all non-trivial methods
- [ ] Added supporting annotations (like @XStreamAlias / @ComponentProfile)
- [ ] Added DefaultValue annotation when there is a default value (e.g. @DefaultValue('true'))
- [ ] Added validation annotation (@NotNull...) when required and add @Valid when nested class has some validation
- [ ] Checked that @NotNull and @NotBlank annotations have a meaningful message when appropriate.
- [ ] Checked that new 3rd party dependencies are appropriately licensed
- [ ] Added comments explaining the "why" and the intent of the code wherever it would not be obvious for an unfamiliar reader
- [ ] Added unit tests or modified existing tests to cover new code paths
- [ ] Tested new/updated components in the UI and at runtime in an Interlok instance
- [ ] Reviewed java class members so that missing annotations are added (InputFieldDefault/ComponentProfile etc)
- [ ] Checked that javadoc generation doesn't report errors
- [ ] Checked the display of the component in the UI
- [ ] Remove any config/license annotations
- [ ] Check the gradle build file to make sure the dependencies section is more explicit "implementation/api".

## Result

What's the end result for the user

## Testing

How can I test this if I'm reviewing this.

telus-terchu committed:
Update libraries to latest

telus-terchu committed:
Update Kinesis using AWS SDK2

sebastien-belin-adp reviewed (changes requested):
@telus-terchu 

Thanks for the PR and welcome to the Interlok team.
I have a few comments below:

- interlok-aws-kinesis and interlok-aws-kinesis-sdk should not change to the `com.adaptris.aws2.kinesis` package but should remain on `com.adaptris.aws.kinesis`.
- We tend to create feature branches like: feature/branch-name
- I have to admin it's a bit difficult for this one but it's nice to add some details of why we are doing the PR, the changed made in the PR, how to test it ... All is in the default PR descirption for next time
- It seems that AWS stopped using the builder API but it seems they switched to a fluent API.
So I guess instead of 

```java
  PutRecordsRequestEntry putRecordsRequestEntry  = new PutRecordsRequestEntry();
  putRecordsRequestEntry.setData(ByteBuffer.wrap(message.getPayload()));
  putRecordsRequestEntry.setPartitionKey(partitionKey);
  return putRecordsRequestEntry;
```

We could do:

```java
  return new PutRecordsRequestEntry()
	  .withData(ByteBuffer.wrap(message.getPayload()))
	  .withPartitionKey(partitionKey);
```

But the result is the same indeed :)

- Out of curiosity, did you run the docker localstack docker image to see if all is working?
To be honest, I think we haven't ran it in years so I don't know if it still does and if it does with AWS2.

telus-terchu committed:
Change package back to com.adaptris.aws.kinesis
Bump docker-compose.yaml to 2.3.2

telus-terchu commented:
@sebastien-belin-adp 

- I have reverted back to the original package
- OK, will create branch under feature next time
- Will also complete the PR comments next time
- I could not find the API that you refer to without builder (see https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/kinesis/model/PutRecordsRequest.html, https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/kinesis/model/PutRecordsRequestEntry.html), could you advise?
- I have now setup docker and localstack and ran the test using junit.localstack.tests.enabled=true. Had to bump the docker-compose.yml version to 2.3.2

sebastien-belin-adp commented:
> @sebastien-belin-adp
> 
> * I have reverted back to the original package
> * OK, will create branch under feature next time
> * Will also complete the PR comments next time
> * I could not find the API that you refer to without builder (see https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/kinesis/model/PutRecordsRequest.html, https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/kinesis/model/PutRecordsRequestEntry.html), could you advise?
> * I have now setup docker and localstack and ran the test using junit.localstack.tests.enabled=true. Had to bump the docker-compose.yml version to 2.3.2

@terry Thanks,

Forget about my comment about the builder. I got confused, they had the fluent API and now have the builder and you to did use the builder so that's great.

telus-terchu committed:
Change package back to com.adaptris.aws.kinesis
Implement createIfNotExists for KinesisSDKStreamProducer

telus-terchu committed:
Update doAwaitStreamActive() to use DescribeStreamRequest.withStreamName()

telus-terchu committed:
Revert incorrect AWS SK update for v1 modules
Revert accidental package renaming

sebastien-belin-adp commented on interlok-aws2-kinesis-sdk/src/main/java/com/adaptris/aws2/kinesis/AWSKinesisSDKConnection.java:
```diff
@@ -0,0 +1,98 @@
+@XStreamAlias("aws-kinesis-sdk-connection")
+@AdapterComponent
+@ComponentProfile(summary = "Connection for supporting connectivity to Amazon Kinesis", tag = "connections,amazon,aws,kinesis",
+    since = "3.12.1")
```
That will be: since 5.0.4

telus-terchu replied:
@sebastien-belin-adp this has been updated

sebastien-belin-adp commented on interlok-aws-kinesis-sdk/src/main/java/com/adaptris/aws/kinesis/KinesisSDKStreamProducer.java:
```diff
@@ -138,12 +174,71 @@ protected void doProduce(AdaptrisMessage msg, String endpoint) throws ProduceExc
-    log.trace("PutRecordResults: {}", putRecordsResult);
+  // According to the docs, we need to wait for the stream to be active
+  // https://docs.aws.amazon.com/streams/latest/dev/kinesis-using-sdk-java-create-stream.html
+  private void doAwaitStreamActive(AmazonKinesis kinesisClient, String endpoint) throws ProduceException {
```
@telus-terchu Most of the line below are not covered by tests.
How much effort would it be to cover them, only if relevant?
Same for the same class in aws2.

telus-terchu replied:
@sebastien-belin-adp I've added a test which covers the majority of the added code.

sebastien-belin-adp replied:
Brillant, thanks

telus-terchu committed:
Update @since for AWSKinesisSDKConnection
Introduce testCreateIfNotExists()
Allow setting max wait time and poll time for KinesisSDKStreamProducer
